packages/action/src/run.ts

```diff
@@ -123,7 +134,7 @@ export async function run() {
 
   if (useMerge && pullRequest?.state == 'open') {
     ref = `refs/pull/${pullRequest.number}/merge`;
-    workspace = undefined;
+    workspace = process.env.GITHUB_WORKSPACE;
```

Comment:
Unfortunately it can't be like that :) The thing is, the `/merge` branch is used to merge the `main` (or any target branch) with PR's branch.
When you do a Pull Request, your branch can be not up to date with `main`, with `refs/pull/{number}/merge`, it's up to date, always.

Here's some context https://github.com/kamilkisiela/graphql-inspector/issues/325